src/invite-api.js

```diff
@@ -0,0 +1,76 @@
+import { TypedEmitter } from 'tiny-typed-emitter'
+import { InviteResponse_Decision } from './generated/rpc.js'
+import { idToKey, keyToId } from './utils.js'
+
+/** @typedef {import('./rpc/index.js').MapeoRPC} MapeoRPC */
+
+export class InviteApi extends TypedEmitter {
+  // TODO: are invites persisted beyond this api?
+  #invites = new Map()
+
+  /**
+   * @param {Object} options
+   * @param {MapeoRPC} options.rpc
+   */
+  constructor({ rpc }) {
+    super()
+    this.rpc = rpc
+
+    // TODO: I'm not seeing encryption keys used in the inviteResponse in the rpc api
+    // what is the purpose of the encryption keys at this stage of the process or afterward?
+    this.rpc.on('invite', (peerId, invite) => {
+      const projectId = keyToId(invite.projectKey)
+      const peerIds = this.#invites.get(projectId) || new Set()
+      peerIds.add(peerId)
+      this.#invites.set(projectId, peerIds)
+
+      if (peerIds.size === 1) {
+        this.emit('invite-received', {
+          projectId,
+          peerId,
+        })
+      }
+    })
+  }
+
+  /**
+   * @param {string} projectId
+   */
+  accept(projectId) {
+    this.#respond({ projectId, decision: InviteResponse_Decision.ACCEPT })
+  }
+
+  /**
+   * @param {string} projectId
+   */
+  reject(projectId) {
+    this.#respond({ projectId, decision: InviteResponse_Decision.REJECT })
+  }
+
+  /**
+   * @param {Object} options
+   * @param {string} options.projectId
+   * @param {InviteResponse_Decision} options.decision
+   */
+  #respond({ projectId, decision }) {
+    const peerIds = this.#getPeerIds(projectId)
+    const projectKey = idToKey(projectId)
+
+    // TODO: should this reply to  one peer with `ACCEPT` and the rest with `ALREADY`?
+    // How is the `ALREADY` decision determined? It looks like that isn't used yet.
+    // Does anything bad happen if we respond to multiple peers with `ACCEPT`?
```

Comment:
Ah good question. The impact would be that each of the invitors receiving `ACCEPT` would write a role record for the invitee. This would be confusing to deal with - we would have duplicate records of a device being added to the project. We should handle this in case it does happen (I've added this to #189).

On reflection I think what would make sense would be to respond `ACCEPT` to one peer (the first that we received an invite from?) and send `ALREADY` to the others.

Reply:
In the case of peers disconnecting, we should send `ACCEPT` to the first still-connected peer that we find.